### PR TITLE
combine markdown and shellcheck to one

### PR DESCRIPTION
**Pull request by hanhsuan**
This could reduce the resource of github action and make the github action faster.

### CODE REVIEW

File: .github/workflows/markdown_and_shellcheck.yml [hunk 1]
```diff
@@ -17,3 +17,8 @@ jobs:
         uses: DavidAnson/markdownlint-cli2-action@v10
         with:
           globs: '**/*.md'
+      - uses: actions/checkout@v3
```
Comment (stanley31huang) [inline .github/workflows/markdown_and_shellcheck.yml:20]: @hanhsuan 
I don't think we need to clone the code again? Is it correct?

Comment (hanhsuan) [reply .github/workflows/markdown_and_shellcheck.yml]: Yes, you are right. I've remove the second one.

**Review by stanley31huang — APPROVED** [review]
LGTM

**Review by baconYao — APPROVED** [review]
LGTM

**Pull request merged**